### PR TITLE
[TR] PIM-5032: Display localized currencies in product grid

### PR DESCRIPTION
**Pull request by pierallard**
| Q                 | A
| ----------------- | ---
| Specs             | y
| Behats            | y
| Blue CI           | c.f. comment
| Changelog updated | y
| Review and 2 GTM  | TODO
| Micro Demo (PO)   | TODO
| Migration script  | n
| Tech Doc          | n

Always failing (on Master too)
features/variant-group/edit_variant_group_attribute_value.feature:66

### CODE REVIEW

File: src/Pim/Bundle/LocalizationBundle/Resources/config/resolvers.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 parameters:
     pim_localization.resolver.locale.class: Pim\Component\Localization\LocaleResolver
+    pim_localization.presenter.price.class: Pim\Component\Localization\PricePresenter
```
Comment (skeleton) [inline src/Pim/Bundle/LocalizationBundle/Resources/config/resolvers.yml]: why declare this presenter in resolvers.yml

File: src/Pim/Component/Localization/PricePresenter.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+use Symfony\Component\Intl\Intl;
+
+/**
+ * Price presenter, able to render price readable for a human
+ *
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class PricePresenter
```
Comment (skeleton) [inline src/Pim/Component/Localization/PricePresenter.php]: no interface ?

File: src/Pim/Component/Localization/PricePresenter.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+{
+    /**
+     * Humanize a price to be readable
+     *
+     * @param array $price   A structured price with keys 'data' and 'currency'
+     * @param array $options The options for presentation
+     *
+     * @return string
+     */
+    public function present($price, $options = [])
```
Comment (skeleton) [inline src/Pim/Component/Localization/PricePresenter.php]: typehint

File: src/Pim/Component/Localization/PricePresenter.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        if (isset($options['locale'])) {
+            $numberFormatter = new \NumberFormatter($options['locale'], \NumberFormatter::CURRENCY);
+
+            return $numberFormatter->formatCurrency($price['data'], $price['currency']);
+        } else {
```
Comment (skeleton) [inline src/Pim/Component/Localization/PricePresenter.php]: else is useless

File: src/Pim/Component/Localization/PricePresenter.php (hunk 2 above)
Comment (skeleton) [inline src/Pim/Component/Localization/PricePresenter.php]: should be in a folder Presenter

File: src/Pim/Component/Localization/PricePresenter.php (hunk 3 above)
Comment (pierallard) [inline src/Pim/Component/Localization/PricePresenter.php]: This signature comes from its interface, should I add `array` ?

**Comment by ClementGautier** [comment]
Add some tests case about negative numbers (do we add the minus symbol before or after the currency ?).

GTM after that

**Comment by pierallard** [comment]
@ClementGautier Done!

File: src/Pim/Component/Localization/Presenter/PricePresenter.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Pim\Component\Localization\Presenter;
+
+use Symfony\Component\Intl\Intl;
+
+/**
+ * Price presenter, able to render price readable for a human
```
Comment (phaseinducer) [inline src/Pim/Component/Localization/Presenter/PricePresenter.php]: Could you be more specific on how this class works please ? (NumberFormatter with fallback on Intl component, etc.)

**Comment by jjanvier** [comment]
gtm

**Comment by phaseinducer** [comment]
GTM

File: src/Pim/Component/Localization/Factory/NumberFactory.php [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        return $formatter;
+    }
+
+    /**
+     * Resolve the options for the factory instances.
+     *
+     * @param array $options
+     * @return array
```
Comment (skeleton) [inline src/Pim/Component/Localization/Factory/NumberFactory.php]: add blank line before @return

File: src/Pim/Component/Localization/Factory/NumberFactory.php [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            [
+                'locale'        => 'en',
+                'type'          => \NumberFormatter::DECIMAL,
+                'number_format' => null
+            ],
+            $options
+        );
+
+        if (null === $options['number_format']
+            && isset($options['locale'])
```
Comment (skeleton) [inline src/Pim/Component/Localization/Factory/NumberFactory.php]: `isset($options['locale'])` is useless
